src/rules/variableNameRule.ts

```diff
@@ -162,6 +176,21 @@ function walk(ctx: Lint.WalkContext<Options>): void {
         }
     }
 
+    function handleVariableDeclaration(node: ts.VariableDeclaration): void {
+        handleDeclaredVariable(node);
+
+        if (!ctx.options.allCapsForConst) {
+            return;
+        }
+
+        const declarationList = node.parent as ts.VariableDeclarationList;
```

Comment:
you can't say for sure if `node.parent` is a VariableDeclarationList. It could also be a CatchClause.
The parent property is already appropriately typed, you only need to assert that it's not undefined